### ISSUE TITLE
What could be the best way to expose soap base endpoint in .net core background application

### ISSUE THREAD

**Issue by KamranShahid** [dotnet/core#4577]
I have a .net core background service application (in .net core 3.1) where I were running some HttpListeners for receiving callbacks getting some callbacks responses. Now one of the client needed a soap base end point as callback.
Please guide me for the best possible way

**Comment by scalablecory** [comment]
A lot of SOAP is fairly easy to parse yourself; you might try using XLinq to get at the relevant data.

@StephenBonikowsky @mconnew do you have any recommendations for exposing SOAP APIs?

**Comment by KamranShahid** [comment]
> A lot of SOAP is fairly easy to parse yourself; you might try using XLinq to get at the relevant data.
> 
> @StephenBonikowsky @mconnew do you have any recommendations for exposing SOAP APIs?

I am asking about exposing soap base service. like WCF service in .net core. I have a third party standard soap wsdl which i need to implement for my callbacks

**Comment by mconnew** [comment]
@scalablecory, have you checked out [CoreWCF](https://github.com/CoreWCF/CoreWCF)?

**Comment by KamranShahid** [comment]
> @KamranShahid, have you checked out [CoreWCF](https://github.com/CoreWCF/CoreWCF)?
> 
> Edit: Added the correct mention

No. One more problem in using soapcore i faced that the thirda party wsdl (which i need to implement) is on soap1.2. so for that i have to support wshttpbinding. but running it in soapcore is giving me platform not supported exception.
Will try this with CoreWCF tomorrow.
Thanks

**Comment by mconnew** [comment]
NetHttpBinding uses SOAP 1.2.  You don't need WSHttpBinding for that.

**Comment by KamranShahid** [comment]
> NetHttpBinding uses SOAP 1.2. You don't need WSHttpBinding for that.

then don't know how to get rid of the error as mentioned at https://github.com/DigDes/SoapCore/issues/457

**Comment by KamranShahid** [comment]
> NetHttpBinding uses SOAP 1.2. You don't need WSHttpBinding for that.

CoreWCF is way behing soapcore. ( https://github.com/CoreWCF/CoreWCF/issues/128)
My problem in soapcore is i have to integrate quite complex wsdl which it self have inner wsdla nd number of reference xsd's with simple and complex types. 
I have logged the issue at https://github.com/DigDes/SoapCore/issues/461
Haven't find anything else other then soapcore and CoreWCF
